test/gurka/test_filter.cc

```diff
@@ -0,0 +1,108 @@
+#include "baldr/graphreader.h"
+#include "mjolnir/util.h"
+
+#include "gurka.h"
+#include "test/test.h"
+
+#if !defined(VALHALLA_SOURCE_DIR)
+#define VALHALLA_SOURCE_DIR
+#endif
+
+using namespace valhalla;
+using namespace valhalla::baldr;
+using namespace valhalla::gurka;
+
+valhalla::gurka::map the_map = {};
+
+class FilterData : public ::testing::Test {
+protected:
+  valhalla::gurka::map BuildPBF(const std::string& workdir) {
+    constexpr double gridsize_metres = 100;
+
+    const std::string ascii_map = R"(
+		A--------B--------C
+                 |
+                 |
+                 |
+                 D
+    )";
+
+    const gurka::ways ways = {
+        {"ABC",
+         {{"highway", "residential"},
+          {"name:forward", "name in forward dir"},
+          {"name:backward", "name in backward dir"},
+          {"osm_id", "100"}}},
+        {"BD", {{"highway", "footway"}}},
```

Comment:
you forgot to draw this one (i know it doesnt matter but its nice that the picture matches the code)

Reply:
I thought it's a laptop viewport issue, since there is a node D. is that way's purpose to break up the ABC way? never looked into the graph filtering, not even sure what it's doing, but probably related to this?